sdks/unity/AgonesSdk.cs

```diff
@@ -0,0 +1,239 @@
+        {
+            return await SendRequestAsync("/allocate", "{}");
+        }
+
+        /// <summary>
+        /// Set a metadata label that is stored in k8s
+        /// </summary>
+        /// <param name="key">label key</param>
+        /// <param name="value">label value</param>
+        /// <returns>
+        /// A task that represents the asynchronous operation.
+        /// The task result contains that the request is success or failure.
+        /// </returns>
+        public async Task<bool> SetLabel(string key, string value)
+        {
+            string json = JsonUtility.ToJson(new KeyValueMessage(key, value));
+            return await SendRequestAsync("/metadata/label", json, UnityWebRequest.kHttpVerbPUT);
+        }
+
+        /// <summary>
+        /// Set a metadata annotation that is stored in k8s
+        /// </summary>
+        /// <param name="key">annotation key</param>
+        /// <param name="value">annotation value</param>
+        /// <returns>
+        /// A task that represents the asynchronous operation.
+        /// The task result contains that the request is success or failure.
+        /// </returns>
+        public async Task<bool> SetAnnotation(string key, string value)
+        {
+            string json = JsonUtility.ToJson(new KeyValueMessage(key, value));
+            return await SendRequestAsync("/metadata/annotation", json, UnityWebRequest.kHttpVerbPUT);
+        }
+        #endregion
+
+        #region AgonesRestClient Private Methods
+        void Health()
+        {
+            _ = SendRequestAsync("/health", "{}");
+        }
+
+        async Task<bool> SendRequestAsync(string api, string json, string method = UnityWebRequest.kHttpVerbPOST)
+        {
+            // To prevent that an async method leaks after destroying this gameObject
+            cancellationTokenSource.Token.ThrowIfCancellationRequested();
+
+            var req = new UnityWebRequest(sidecarAddress + api, method)
+            {
+                uploadHandler = new UploadHandlerRaw(Encoding.UTF8.GetBytes(json)),
+                downloadHandler = new DownloadHandlerBuffer()
+            };
+            req.SetRequestHeader("Content-Type", "application/json");
+
+            await new AgonesAsyncOperationWrapper(req.SendWebRequest());
+
+            bool ok = req.responseCode == 200;
+
+            if (ok)
+            {
+                Log($"Agones SendRequest ok: {api} {req.downloadHandler.text}");
+            }
+            else
+            {
+                Log($"Agones SendRequest failed: {api} {req.error}");
+            }
+
+            return ok;
+        }
+
+        void Log(object message)
+        {
+            if (!logEnabled) { return; }
+
+#if UNITY_EDITOR
+            Debug.Log(message);
+#else
+            Console.WriteLine(message);
+#endif
+        }
+        #endregion
+
+        #region AgonesRestClient Nested Classes
+        class AgonesAsyncOperationWrapper

```

Comment:
Please add `<summary>` class documentation to these classes and their public fields/methods.